scheduler/manager/cdn_manager.go

```diff
@@ -217,25 +246,24 @@ func (cm *CDNManager) processPieceSeed(task *types.Task, ps *cdnsystem.PieceSeed
 	hostId := cm.getHostUuid(ps)
 	host, ok := cm.hostManager.Get(hostId)
 	if !ok {
-		ip, rpcPort, downPort := "", 0, 0
-		cdnInfo := cm.getCdnInfo(ps.SeederName)
-		if cdnInfo != nil {
-			ip, rpcPort, downPort = cdnInfo.IP, int(cdnInfo.RpcPort), int(cdnInfo.DownloadPort)
-		} else {
+		server, found := cm.getServer(ps.SeederName)
+		if !found {
 			logger.Errorf("get cdn by SeederName[%s] failed", ps.SeederName)
 		}
+
 		host = &types.Host{
 			Type: types.HostTypeCdn,
 			PeerHost: scheduler.PeerHost{
 				Uuid:     hostId,
 				HostName: ps.SeederName,
-				Ip:       ip,
-				RpcPort:  int32(rpcPort),
-				DownPort: int32(downPort),
+				Ip:       server.HostInfo.Ip,
```

Comment:
if server not found, server.HostInfo.Ip  will panic

Reply:
It is impossible for the server to exist but hostInfo does not exist.